redis-om-spring/src/main/java/com/redis/om/spring/RedisEnhancedKeyValueAdapter.java

```diff
@@ -261,32 +321,24 @@ public void deleteAllOf(String keyspace) {
    * @param type     the desired target type.
    * @param offset   index value to start reading.
    * @param rows     maximum number or entities to return.
-   * @param <T>      type of entity
+   * @param <T>
    * @return never {@literal null}.
+   * @since 2.5
    */
   @Override
   public <T> List<T> getAllOf(String keyspace, Class<T> type, long offset, int rows) {
-    byte[] binKeyspace = toBytes(keyspace);
-
-    Set<byte[]> ids = redisOperations
-        .execute((RedisCallback<Set<byte[]>>) connection -> connection.sMembers(binKeyspace));
-
-    List<T> result = new ArrayList<>();
-    List<byte[]> keys = new ArrayList<>(ids);
-
-    if (keys.isEmpty() || keys.size() < offset) {
-      return Collections.emptyList();
-    }
-
-    offset = Math.max(0, offset);
-    if (rows > 0) {
-      keys = keys.subList((int) offset, Math.min((int) offset + rows, keys.size()));
-    }
-
-    for (byte[] key : keys) {
-      result.add(get(key, keyspace, type));
+    List<String> keys = getAllIds(keyspace, type);
```

Comment:
It looks like there's an extra step here, It seems to me you're running:

```
FT.SEARCH index * RETURN 1 id LIMIT offset rows
```

and then a sequence of:

```
HGETALL id
```

You could just short circuit the whole thing and just use:

```
FT.SEARCH index * LIMIT offset rows
```